minecraft-plugin/src/main/java/com/pvchat/proximityvoicechat/plugin/distanceMatrix/PlayerDistanceAndVolumeCalculations.java

```diff
@@ -0,0 +1,122 @@
+    public List<Player> players;
+    private ProximityVoiceChat pluginInstance;
+    private HashMap<String, List<Player>> playersToWorld;
+    private ArrayList<Consumer<List<PlayerVolumeData>>> stateChangeListeners;
+
+    public PlayerDistanceAndVolumeCalculations(ProximityVoiceChat pluginInstance) {
+        this.pluginInstance = pluginInstance;
+        players = new ArrayList<>();
+        playersToWorld = new HashMap<>();
+        stateChangeListeners = new ArrayList<>();
+    }
+
+    public void addStateChangeListener(Consumer<List<PlayerVolumeData>> stateChangeListener){
+        stateChangeListeners.add(stateChangeListener);
+    }
+
+    public void removeStateChangeListener(Consumer<List<PlayerVolumeData>> stateChangeListener) {
+        stateChangeListeners.remove(stateChangeListener);
+    }
+
+    //Getting Online players
+    public void getPlayers() {
+        players.clear();
+        players.addAll(Bukkit.getOnlinePlayers().stream().toList());
+
+        playersToWorld.clear();
+
+        for(World w : Bukkit.getServer().getWorlds()){
+            playersToWorld.put(w.getName(), new ArrayList<>());
+        }
+
+        //Dividing per realm
+        for (Player temp : players) {
+            String world = temp.getWorld().getName();
+
+            playersToWorld.get(world).add(temp);
+        }
+    }
+
+    private void addToVolumeList(List<PlayerVolumeData>  playerVolumeList, List<Player>  playersOfSomeWorld){
+        int size = playersOfSomeWorld.size();
+        if (size > 1) {
+            for (int i = 0; i < size; i++) {
+                for (int j = i + 1; j < size; j++) {
+                    Player p1, p2;
+                    p1 = playersOfSomeWorld.get(i);
+                    p2 = playersOfSomeWorld.get(j);
+                    double distance = distanceCalculator(p1.getLocation(), p2.getLocation());
+                    long volume = calculateVolume(distance);
+                    if (volume != -1) {
+                        PlayerVolumeData temporary = new PlayerVolumeData(p1.getUniqueId().toString(),p2.getUniqueId().toString(),volume);
+                        playerVolumeList.add(temporary);
+                    }
+                }
+            }
+        }
+    }
+
+    //getting list of player volume pairs
+    public List<PlayerVolumeData> playerVolumeList() {
+        ArrayList<PlayerVolumeData> p = new ArrayList<>();
+        if (players.size() > 1) {
+            for (Map.Entry<String,List<Player>> entry : playersToWorld.entrySet()){
+                addToVolumeList(p, entry.getValue());
+            }
+
+        }
+        if (p.isEmpty()) {
+            return null;
+        }
+        return p;
+    }
+
+
+    //Calculating Distance
+    private double distanceCalculator(Location location1, Location location2) {
+        return location1.distance(location2);
+    }
+
+    //Calculating Volume
+    private long calculateVolume(double distance) {
+        if (distance > 110) {
+            return -1;
```

Comment:
I use the result of this function in the "addToVolumeList" function to determine who apears on the list so if the result is "-1" then the players dont appear on the list. As to whether its "-1" or "0" its moustly personal preference, but i think its safer result because calculated volume can't be negative so  returning negative value is easyly recognizable as signal thet two players dont hear each other.